### ISSUE TITLE
Password protected pdf generate?

### ISSUE THREAD

**Issue by manprajapat** [marcbachmann/node-html-pdf#120]
By using this module can we make pdf files password protected?

**Comment by marcbachmann** [comment]
No, sorry, doesn't work.

**Comment by manprajapat** [comment]
ok. can you suggest me anything else to generate password protect pdf file in node js.

**Comment by marcbachmann** [comment]
No, don't know any module. But pdf protection isn't secure anyways. I prefer to protect something the right way by serving something with a server and grant access using a token.

Sorry, that this is not really the answer you expected.
I hope that's not a business requirement you have :smile:

**Comment by manprajapat** [comment]
ok. :) It seems that you are putting it in next versions?

**Comment by marcbachmann** [comment]
>  It seems that you are putting it in next versions?

No

**Comment by nrhirani** [comment]
@manprajapat: You can use [node-qpdf](https://github.com/nrhirani/node-qpdf) on the second pass, i.e after creating PDF through `node-html-pdf`.  I was facing similar issue so went ahead and created a wrapper around [qpdf](http://qpdf.sourceforge.net/) which supports password protect. It supports AES 256, AES 128, RC4 128 & RC4 40 all supported by Adobe Reader.

**Comment by manprajapat** [comment]
how we can install qpdf in node js?
can you tell me the process after downloading it from sourceforge?

**Comment by vjoshihumancare** [comment]
We have similar issue

**Comment by georgeenathomas** [comment]
@nrhirani Could you please share the wrapper you created? I am facing a similar issue and would be helpful
